services/user/badge.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package user
+
+import (
+	"context"
+	"fmt"
+
+	"code.gitea.io/gitea/models/db"
+	user_model "code.gitea.io/gitea/models/user"
+)
+
+// UpdateBadgeDescription changes the description and/or image of a badge
+func UpdateBadge(ctx context.Context, b *user_model.Badge) error {
+	return db.WithTx(ctx, func(ctx context.Context) error {
+		return user_model.UpdateBadge(ctx, b)
+	})
+}
+
+// DeleteBadge remove record of badge in the database
+func DeleteBadge(ctx context.Context, b *user_model.Badge) error {
+	return db.WithTx(ctx, func(ctx context.Context) error {
+		if err := user_model.DeleteBadge(ctx, b); err != nil {
+			return fmt.Errorf("DeleteBadge: %w", err)
+		}
+		return nil
+	})
```

Comment:
There is already a transaction in `DeleteBadge`.

So why we need this wrap function in service package?